magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -0,0 +1,571 @@
+        for j in list_of_runs[0:split_percent]:
+            os.system(
+                f"mv {j} {proton_dir}/train/{list_of_dir[directory].split('/')[-2]}"
+            )
+
+        os.system(
+            f"cp {list_of_dir[directory]}*.txt {proton_dir}/train/{list_of_dir[directory].split('/')[-2]}"
+        )
+        os.system(
+            f"mv {list_of_dir[directory]}*.txt {proton_dir}/../protons_test/{list_of_dir[directory].split('/')[-2]}"
+        )
+        os.system(
+            f"mv {list_of_dir[directory]}*.h5 {proton_dir}/../protons_test/{list_of_dir[directory].split('/')[-2]}"
+        )
+        os.system(f"rm -r {list_of_dir[directory]}")
+
+
+def merge(target_dir, identification, MAGIC_runs, env_name, source, NSB_match):
+
+    """
+    This function creates the bash scripts to run merge_hdf_files.py in all MAGIC subruns.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    identification : str
+        Tells which batch to create. Options: subruns, M1M2, nights
+    MAGIC_runs : matrix of strings
+        This matrix is imported from config_general.yaml and tells the function where to find the data and where to put the merged files
+    env_name : str
+        Name of the environment
+    source : str
+        Target name
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    process_name = f"merging_{source}"
+    if not NSB_match:
+        MAGIC_DL1_dir = f"{target_dir}/{source}/DL1/Observations"
+        if os.path.exists(f"{MAGIC_DL1_dir}/M1") & os.path.exists(
+            f"{MAGIC_DL1_dir}/M2"
+        ):
+            if not os.path.exists(f"{MAGIC_DL1_dir}/Merged"):
+                os.mkdir(f"{MAGIC_DL1_dir}/Merged")
+
+        with open(f"{source}_Merge_MAGIC_{identification}.sh", "w") as f:
+            lines = [
+                "#!/bin/sh\n\n",
+                "#SBATCH -p short\n",
+                f"#SBATCH -J {process_name}\n",
+                "#SBATCH -n 1\n\n",
+                f"#SBATCH --output={MAGIC_DL1_dir}/Merged/slurm-%x.%j.out"
+                f"#SBATCH --error={MAGIC_DL1_dir}/Merged/slurm-%x.%j.err"
+                "ulimit -l unlimited\n",
+                "ulimit -s unlimited\n",
+                "ulimit -a\n\n",
+            ]
+            f.writelines(lines)
+
+            if identification == "0_subruns":
+                if os.path.exists(f"{MAGIC_DL1_dir}/M1"):
+                    for i in MAGIC_runs:
+                        if not os.path.exists(f"{MAGIC_DL1_dir}/Merged/{i[0]}"):
+                            os.mkdir(
+                                f"{MAGIC_DL1_dir}/Merged/{i[0]}"
+                            )  # Creating a merged directory for the respective night
+                        if not os.path.exists(f"{MAGIC_DL1_dir}/Merged/{i[0]}/{i[1]}"):
+                            os.mkdir(
+                                f"{MAGIC_DL1_dir}/Merged/{i[0]}/{i[1]}"
```

Comment:
os.makedirs will allow you to skip if, and also the previous two os.mkdir on the two parent directories